### PR TITLE
Respect defaultApp settting even if there is multiroot or no workspace

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/microsoft/vscode-azureappservice/issues/748

It also will work for a multiroot workspace and only respect/reset the setting in the appropriate workspaces.

Also I'll move the IDeployWizardContext to its own file, but other PR also changes it so will do it in a separate PR

### CODE REVIEW

File: src/commands/deploy.ts [hunk 1]
```diff
@@ -23,64 +22,40 @@ import { isPathEqual, isSubpath } from '../utils/pathUtils';
 import { getRandomHexString } from "../utils/randomUtils";
 import * as workspaceUtil from '../utils/workspace';
 import { cancelWebsiteValidation, validateWebSite } from '../validateWebSite';
+import { getDefaultWebAppToDeploy } from './getDefaultWebAppToDeploy';
 import { startStreamingLogs } from './startStreamingLogs';
 
 // tslint:disable-next-line:max-func-body-length cyclomatic-complexity
```
Comment (EricJizbaMSFT) [inline src/commands/deploy.ts]: I like moving `getDefaultWebAppToDeploy` to its own file - does that mean we enable either or both of these rules now? 😄

File: src/commands/deploy.ts [hunk 2]
```diff
@@ -154,38 +131,46 @@ export async function deploy(context: IDeployWizardContext, confirmDeployment: b
...
         if (result === resetDefault) {
-            const localRootPath = nonNullValue(currentWorkspace).uri.fsPath;
-            const settingsPath = path.join(localRootPath, '.vscode', 'settings.json');
-            const doc = await workspace.openTextDocument(Uri.file(settingsPath));
-            window.showTextDocument(doc);
-            await workspaceConfig.update(constants.configurationSettings.defaultWebAppToDeploy, '');
+            if (context.configurationTarget === ConfigurationTarget.Global) {
+                await commands.executeCommand('workbench.action.openGlobalSettings');
+            } else {
+                const localRootPath = nonNullValue(currentWorkspace).uri.fsPath;
```
Comment (EricJizbaMSFT) [inline src/commands/deploy.ts]: Can you change this case to use a command as well? `'workbench.action.openSettings'` maybe?

File: src/commands/deploy.ts (hunk 1 above)
Comment (nturinski) [reply src/commands/deploy.ts]: There's still quite a bit of work to do on that front. . .

image

😓

Comment (EricJizbaMSFT) [reply src/commands/deploy.ts]: Oof

File: src/commands/deploy.ts (hunk 2 above)
Comment (nturinski) [reply src/commands/deploy.ts]: Using `workbench.action.openSettingsJson` seems to just open the User settings as well whether I have a single root or multiroot workspace 😕

**Pull request merged**